backend/src/data_classes/Connection.py

```diff
@@ -3,18 +3,40 @@
 from datetime import datetime, timedelta
 from typing import List
 
+from src.config import DATETIME_FORMAT
 from src.data_classes.Transfer import Transfer
 from src.data_classes.Walk import Walk
 from src.data_classes.IAction import IAction
 
 
 @dataclass
 class Connection:
+    walking_only: bool
+    start_stop_name: str
+    end_stop_name: str
+    start_datetime: datetime
+    end_datetime: datetime
     transfers: List[IAction]
 
     def __str__(self) -> str:
         return '\n'.join(map(str, self.transfers))
 
+    def __init__(self, a_list: List[IAction]):
+        if len(a_list) == 1 and type(a_list[0]) == Walk:
+            self.walking_only = True
+            self.transfers = a_list
+            self.start_stop_name = None
+            self.end_stop_name = None
+            self.start_datetime = None
+            self.end_datetime = None
+        else:
+            self.walking_only = False
+            self.start_stop_name = a_list[0].start_stop_name if type(a_list[0]) != Walk else a_list[1].start_stop_name
+            self.end_stop_name = a_list[-1].end_stop_name if type(a_list[-1]) != Walk else a_list[-2].end_stop_name
+            self.start_datetime = a_list[0].start_datetime if type(a_list[0]) != Walk else a_list[1].start_datetime
+            self.end_datetime = a_list[-1].end_datetime if type(a_list[-1]) != Walk else a_list[-2].end_datetime
+            self.transfers = a_list
+
```

Comment:
`Connection` wraps a sequence of `IActions`.

`walking_only` should be a dynamic property (`@property`).

`self.transfers` is of type `List[IAction]`, so `Walk` cannot be directly assigned.

`isinstance(x, Class)` is usually preferred over `type(x) == Class` (imagine `class Run` extending `Walk`)